src/full_node/full_node.py

```diff
@@ -392,7 +379,8 @@ async def _sync(self):
                 return
 
             fork_point_height = self.sync_store.get_potential_fork_point(heaviest_peak_hash)
-
+            # cache warmup to fork height
+            await self.blockchain.forkpoint_warmup(fork_point_height)
```

Comment:
We need to warmup after syncing also. (even if there was an exception in sync). Otherwise we will be in an invalid state after syncing

Reply:
why ? you are adding blocks to the cache as you move higher with the sync, you should have peak-10000 in the cache after sync, if sync fails then there will be a new sync and the warmup will be triggered there